CMakeLists.txt

```diff
@@ -179,6 +179,12 @@ option(POCL_ICD_ABSOLUTE_PATH "Use absolute path in pocl.icd" ON)
 
 option(ENABLE_POCL_BUILDING "When OFF, env var POCL_BUILDING has no effect. Defaults to ON" ON)
 
+option(VISIBILITY_HIDDEN "Build with -fvisibility=hidden -fvisibility-inlines-hidden" OFF)
```

Comment:
Is there a reason to not build with this mode? I prefer to avoid additional build options (especially default OFF ones) since they tend to bitrot easily.

Reply:
If it works with all configurations it would be nice to enable it by default and even drop the switch ;-)

Reply:
I pushed it but now that I tried to actually enable this, I'm getting 
```libpocl-devices-cuda.so:
  undefined symbol: pocl_get_string_option```
with my CUDA environment.

Reply:
As I wrote, so far I only tested it for the cpu devices and made the symbol set minimal for them.

Can you paste `nm -D lib/CL/devices/XXX/libpocl-devices-XXX.so` for the non-CPU devices from a "classic" build and I'll see what symbols are missing an `POCL_EXPORT`?

Reply:
https://termbin.com/hju9

Reply:
A new commit is available in my `visibility-hidden` branch that should fix that for the `cuda` device, but we need to check the `accel`, `hsa` and `tce` devices, too.
Any hints on how `libllvmopencl.so` is to be used?

Reply:
I do not think libllvmopencl.so is used as a separate library any longer. We used to call it from opt command line but it was long time ago, and I believe this is not done anymore anywhere. Could you just try removing the build of the .so and see if any of the tests explode?

Reply:
Disabling `libllvmopencl.so` did not show any bad effects, the corresponding commit is on top of my `visibility-hidden` branch.
The last usage reference I could find was `scripts/pocl-standalone.in` which is gone since 0.14.